Assginment-1 (Calculator)

khushivj711 opened the pull request:
Design a console-based calculator program in C that accepts mathematical expressions as input in the form of strings.
The program should evaluate the expression and return the result. Supported operations include addition (+), subtraction (-), multiplication (*), and division (/).
The program should handle integer operations and output the result as an integer, even if the result of division has a remainder.

khushivj711 committed:
assignment -1 of calculator.

badal-soni commented on calculator.c:
```diff
@@ -0,0 +1,133 @@
+
+int solveExpr(const char *expr, int *err)
+{
+    int result = solveTerm(&expr, err);
```
avoid using recursion: solveExpr -> solveTerm -> solveExpr. Please come-up with an interative solution.

badal-soni commented on calculator.c:
```diff
@@ -0,0 +1,133 @@
+
+    while (*expr == '+' || *expr == '-')
+    {
+        char op = *expr;
```
use descriptive names, 'op' is not describing anything

badal-soni commented on calculator.c:
```diff
@@ -0,0 +1,133 @@
+
+int main()
+{
+    char input[100];
```
use an infinite loop and give an option to user to break out from the loop. Also keep main function small (move logic from main to a separate function)

badal-soni commented on calculator.c:
```diff
@@ -0,0 +1,133 @@
+    {
+        char op = **expr;
+        (*expr)++;
+        int right = solveFactor(expr, err);
```
what is 'right' ?

badal-soni reviewed (changes requested):
Explore:
1. how to avoid magic literals in code
2. how to modularize and reuse the code

khushivj711 committed:
Update calculator.c

i tried to use the iterative approach , meaningfull variables and functions name , main() function of 2 lines and more etc.

badal-soni commented on calculator.c:
```diff
@@ -0,0 +1,156 @@
+      pushNumber(number);
+      continue;
+    }
+    if (expression[index] == '+' || expression[index] == '-' ||
```
extract out this complete condition in a separate function which should return boolean: isOperator(char ch)

khushivj711 committed:
Update calculator.c

write the isOperator(char) function to modularize

badal-soni commented:
Please merge this PR
